### PR TITLE
feat: implement limit to last feature

### PR DESCRIPTION
**Pull request by IlyaFaer**
Here is what Ben Whitehead said about this feature:

> limitToLast is a client side only feature that changes the orientation of the cursor in the query so that it inverts the index being scanned.
> 
> So `select docs order by doc.updated_timestamp ASC limitToLast 10` becomes `select docs order by doc.updated_timestamp DESC limit 10` with a client side traversal inversion. Because of this, the client has to buffer all of the documents before returning the query results, as such it's not possible to stream(). 
> 
> If we need to de-deprecate the get() method to allow this type of query for customers I think that is okay from a feature standpoint.

See PR 954 in nodejs-firestore and PR 151 in java-firestore to know how googlers implemented the feature in other languages.

### CODE REVIEW

File: google/cloud/firestore_v1/query.py [hunk 1]
```diff
@@ -729,13 +761,35 @@ def _to_protobuf(self):
...
+        """
+        if self._limit_to_last:
+            self._limit = self._limit_to_last
+
+            for order in self._orders:
+                order.direction = _enum_from_direction(
+                    self.DESCENDING
+                    if order.direction == self.ASCENDING
+                    else self.ASCENDING
+                )
```
Comment (IlyaFaer) [inline google/cloud/firestore_v1/query.py:807]: Flipping the order before sending the query to backend

File: google/cloud/firestore_v1/query.py [hunk 2]
```diff
@@ -729,13 +761,35 @@ def _to_protobuf(self):
...
+
+        result = list(self.stream(transaction=transaction))
+        return list(reversed(result))
```
Comment (IlyaFaer) [inline google/cloud/firestore_v1/query.py]: We can't use `sort()` in here, 'cause there is no comparison methods in document objects. So we're using `reversed()` to create a reversed iterator, and converting it to `list()` before return.

File: tests/unit/v1/test_query.py [hunk 3]
```diff
@@ -1104,9 +1109,57 @@ def test_get_simple(self):
...
+        returned = list(get_response)
+        self.assertEqual(len(returned), 2)
+
+        snapshot = returned[0]
+        self.assertEqual(snapshot.reference._path, ("dee", "sleep"))
+        self.assertEqual(snapshot.to_dict(), data2)
+
+        snapshot2 = returned[1]
+        self.assertEqual(snapshot2.reference._path, ("dee", "sleep"))
+        self.assertEqual(snapshot2.to_dict(), data)
```
Comment (IlyaFaer) [inline tests/unit/v1/test_query.py]: Checking that data came in reverse order, as limit to last is a limit made from the end

File: google/cloud/firestore_v1/query.py [hunk 4]
```diff
@@ -729,13 +761,35 @@ def _to_protobuf(self):
...
-        """Deprecated alias for :meth:`stream`."""
-        warnings.warn(
-            "'Query.get' is deprecated:  please use 'Query.stream' instead.",
-            DeprecationWarning,
-            stacklevel=2,
-        )
-        return self.stream(transaction=transaction)
+        """Read the documents in the collection that match this query.
+
+        This sends a ``RunQuery`` RPC and returns a list with documents
```
Comment (mf2199) [inline google/cloud/firestore_v1/query.py]: `a list of documents`

File: google/cloud/firestore_v1/query.py (hunk 2 above)
Comment (mf2199) [reply google/cloud/firestore_v1/query.py]: :+1:  Can we propose a comparison method in a separate PR, what do you think?

File: tests/unit/v1/test_query.py (hunk 3 above)
Comment (mf2199) [reply tests/unit/v1/test_query.py]: Just to make sure it's working as advertized, maybe it's worth adding another method to the system test as well.

**Review by mf2199 — COMMENTED** [review]
I also noticed possible inconsistencies in the docstrings. For example,
```
projection (Optional[:class:`google.cloud.proto.firestore.v1.\
```
should probably read
```
projection (Optional[:class:`google.cloud.firestore_v1.proto.\
```
These can be addressed in a separate PR.

File: google/cloud/firestore_v1/query.py (hunk 4 above)
Comment (IlyaFaer) [reply google/cloud/firestore_v1/query.py]: Done

File: google/cloud/firestore_v1/query.py (hunk 2 above)
Comment (IlyaFaer) [reply google/cloud/firestore_v1/query.py]: I'll ask Ben. I don't think they actually want it, as I saw some time ago a PR from a user with such a thing, and they closed it

**Comment by IlyaFaer** [comment]
Merged into Google

**Pull request closed without merging**